crates/matrix-sdk/src/widget/mod.rs

```diff
@@ -0,0 +1,52 @@
+//! Client widget API implementation.
+
+use tokio::sync::mpsc::{Receiver, Sender};
+
+use crate::room::Room as JoinedRoom;
+
+mod permissions;
+
+pub use self::permissions::{Permissions, PermissionsProvider};
+
+/// Describes a widget.
+#[derive(Debug)]
+pub struct Widget {
+    /// Information about the widget.
+    pub info: Info,
+    /// Communication channels with a widget.
+    pub comm: Comm,
+}
+
+/// Information about a widget.
+#[derive(Debug)]
+pub struct Info {
+    /// Widget's unique identifier.
+    pub id: String,
+    /// Whether or not the widget should be initialized on load message
+    /// (`ContentLoad` message), or upon creation/attaching of the widget to
+    /// the SDK's state machine that drives the API.
+    pub init_on_load: bool,
```

Comment:
I think we should call it `init_on_content_loaded` even with the doc text I can see how one interprets it as:
init_on_load -> the widget initializes when the I frame is attached/created/"loaded".
(But we want this bool to define define, that the widget is initialized once it sends us the content loaded action.